### PR TITLE
[PR2] [Channel] Generate ci_channel, co_channel, di_channel, do_channel class files

### PR DESCRIPTION
**Pull request by samchris007**
### What does this Pull Request accomplish?
- Used the metadata (**attributes,py**) from **PR:** https://github.com/ni/grpc-device-scrapigen/pull/148 to generate channel class files.
- This PR includes generation of channel class files such as,
  - **ci_channel.py** 
  - **co_channel.py** 
  - **di_channel.py** 
  - **do_channel.py** 
- In **ci_channel.py**, name of the properties that are changed are,
<html></head><body link="#0563C1" vlink="#954F72">

Current Name | Changed To
-- | --
ci_count_edges_count_reset_reset_cnt | ci_count_edges_count_reset_reset_count
ci_dup_count_prevention | ci_dup_count_prevent
ci_pulse_freq_starting_edge | ci_pulse_freq_start_edge
ci_pulse_ticks_starting_edge | ci_pulse_ticks_start_edge
ci_pulse_time_starting_edge | ci_pulse_time_start_edge
ci_velocity_a_input_dig_fltr_enable | ci_velocity_encoder_a_input_dig_fltr_enable
ci_velocity_a_input_dig_fltr_min_pulse_width | ci_velocity_encoder_a_input_dig_fltr_min_pulse_width
ci_velocity_a_input_dig_fltr_timebase_rate | ci_velocity_encoder_a_input_dig_fltr_timebase_rate
ci_velocity_a_input_dig_fltr_timebase_src | ci_velocity_encoder_a_input_dig_fltr_timebase_src
ci_velocity_a_input_logic_lvl_behavior | ci_velocity_encoder_a_input_logic_lvl_behavior
ci_velocity_a_input_term | ci_velocity_encoder_a_input_term
ci_velocity_a_input_term_cfg | ci_velocity_encoder_a_input_term_cfg
</body></html></div>
</body>
</html>

### Why should this Pull Request be merged?
Implements [Task 2316151](https://dev.azure.com/ni/DevCentral/_workitems/edit/2316151): Generate all the channel classes in task modules

### CODE REVIEW

File: src/codegen/metadata/script_info.py [hunk 1]
```diff
@@ -8,5 +8,21 @@
             "relativeOutputPath": "_task_modules\\channels\\ao_channel.py",
             "templateFile": "_task_modules\\channels\\ao_channel.py.mako",
         },
+        {
+            "relativeOutputPath": "_task_modules\\channels\\ci_channel.py", 
+            "templateFile": "_task_modules\\channels\\ci_channel.py.mako"
+        },
+        {
```
Comment (zhindes) [inline src/codegen/metadata/script_info.py:15]: did the output of none of these change? if so, that's a miracle!

File: generated/nidaqmx/_task_modules/channels/ci_channel.py [hunk 2]
```diff
@@ -1211,7 +1211,7 @@ def ci_count_edges_count_reset_logic_lvl_behavior(self):
         check_for_error(error_code)
 
     @property
-    def ci_count_edges_count_reset_reset_cnt(self):
+    def ci_count_edges_count_reset_reset_count(self):
```
Comment (zhindes) [inline generated/nidaqmx/_task_modules/channels/ci_channel.py:1214]: this is fine, but add a deprecation like we did with [ai](https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_task_modules/channels/ai_channel.py#L9379)

File: generated/nidaqmx/_task_modules/channels/ci_channel.py [hunk 3]
```diff
@@ -8111,7 +8111,7 @@ def ci_pulse_freq_logic_lvl_behavior(self):
         check_for_error(error_code)
 
     @property
-    def ci_pulse_freq_starting_edge(self):
+    def ci_pulse_freq_start_edge(self):
```
Comment (zhindes) [inline generated/nidaqmx/_task_modules/channels/ci_channel.py:8114]: I think I'm ok with `start` over `starting`. That is what the C API uses, and it still reads fine. But lets add deprecations, like [ai](https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_task_modules/channels/ai_channel.py#L9379).

File: generated/nidaqmx/_task_modules/channels/ci_channel.py [hunk 4]
```diff
@@ -11746,8 +11896,8 @@ def ci_velocity_a_input_dig_fltr_enable(self):
 
         return val.value
 
-    @ci_velocity_a_input_dig_fltr_enable.setter
-    def ci_velocity_a_input_dig_fltr_enable(self, val):
+    @ci_velocity_encoder_a_input_dig_fltr_enable.setter
```
Comment (zhindes) [inline generated/nidaqmx/_task_modules/channels/ci_channel.py:11899]: these are fine, but add deprecations

**Review by zhindes — CHANGES_REQUESTED** [review]

File: generated/nidaqmx/_task_modules/channels/ci_channel.py [hunk 5]
```diff
@@ -3325,7 +3325,7 @@ def ci_data_xfer_req_cond(self):
         check_for_error(error_code)
 
     @property
-    def ci_dup_count_prevention(self):
+    def ci_dup_count_prevent(self):
```
[the remark this answers is not in the document]
Comment (samchris007) [reply generated/nidaqmx/_task_modules/channels/ci_channel.py]: Updated.

File: generated/nidaqmx/_task_modules/channels/ci_channel.py (hunk 3 above)
Comment (samchris007) [reply generated/nidaqmx/_task_modules/channels/ci_channel.py:8115]: Added deprecation for the `ci_pulse_freq_starting_edge`

File: generated/nidaqmx/_task_modules/channels/ci_channel.py (hunk 4 above)
Comment (samchris007) [reply generated/nidaqmx/_task_modules/channels/ci_channel.py:11900]: Added deprecation for the `ci_velocity_a_input_dig_fltr_enable`

File: generated/nidaqmx/_task_modules/channels/ci_channel.py (hunk 2 above)
Comment (samchris007) [reply generated/nidaqmx/_task_modules/channels/ci_channel.py:1215]: Added deprecation for the `ci_count_edges_count_reset_reset_cnt`

**Review by bkeryan — APPROVED** [review]
I'm deferring to Zach regarding the name changes.

**Review by zhindes — APPROVED** [review]
looks great

**Comment by zhindes** [comment]
@samchris007 @subash-suresh  - since this is going into [users/subash/generate-channel-classes](https://github.com/ni/nidaqmx-python/tree/users/subash/generate-channel-classes), I'll let y'all handle squash and merge. I think you should have permission to do that?

**Comment by zhindes** [comment]
@samchris007 @subash-suresh  - if you do change the target branch to `master`, ping me and let me do the merge. I like to clean up the commit message a bit to remove all the internal changes from code reviews. No big deal, just FYI for the future.

**Pull request merged**